File: suites/01_algorithms/comparisons/prime_sieve.py
import math


def is_prime(n: int) -> bool:
    if n < 2:
        return False
    for d in range(2, int(math.isqrt(n)) + 1):
        if n % d == 0:
            return False
    return True


def prime_count(limit: int) -> int:
    total = 0
    for n in range(2, limit + 1):
        if is_prime(n):
            total += 1
    return total


if __name__ == "__main__":
    print(prime_count(25000))

File: suites/01_algorithms/comparisons/prime_sieve.go
package main

import (
	"fmt"
	"math"
)

func isPrime(n int64) bool {
	if n < 2 {
		return false
	}
	limit := int64(math.Sqrt(float64(n)))
	for d := int64(2); d <= limit; d++ {
		if n%d == 0 {
			return false
		}
	}
	return true
}

func primeCount(limit int64) int64 {
	var total int64
	for n := int64(2); n <= limit; n++ {
		if isPrime(n) {
			total++
		}
	}
	return total
}

func main() {
	fmt.Println(primeCount(25000))
}
